benchmark/sdpa_benchmark_training/benchmark_single_sdpa.py

```
@@ -730,24 +777,76 @@ def convert_to_cudnn_type(torch_type):
+            # Only the measured pass must run on an OSS kernel. For bwd-only
+            # runs the fwd graph is just input prep — leave it fully unpinned,
+            # or a FROST engine whose build fails (e.g. an arch the probe
+            # accepts but the lowering rejects) crashes a case whose measured
+            # pass never needed it.
+            pin_frost_engine_or_skip(graph_fwd, "fwd", required=True)
         graph_fwd.check_support()
-        graph_fwd.build_plans()
+        if args.sdpa_backend == "cudnn_oss" and run_fwd:
+            try:
+                graph_fwd.build_plans()
+            except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)